feature-libs/checkout/base/components/components/shipping-address/shipping-address.component.spec.ts

```diff
@@ -0,0 +1,403 @@
+    component.showNewAddressForm();
+    expect(component.addressFormOpened).toEqual(true);
+  });
+
+  it('should call hideNewAddressForm()', () => {
+    component.hideNewAddressForm();
+    expect(component.addressFormOpened).toEqual(false);
+
+    spyOn(component, 'back');
+    component.hideNewAddressForm(true);
+    expect(component.back).toHaveBeenCalled();
+  });
+
+  it('should be able to go to next step', () => {
+    component.next();
+    expect(checkoutStepService.next).toHaveBeenCalledWith(
+      <any>mockActivatedRoute
+    );
+  });
+
+  it('should be able to go to previous step', () => {
+    component.back();
+    expect(checkoutStepService.back).toHaveBeenCalledWith(
+      <any>mockActivatedRoute
+    );
+  });
+
+  it('should be able to select address', () => {
+    component.selectAddress({});
+    expect(
+      checkoutDeliveryAddressFacade.setDeliveryAddress
+    ).toHaveBeenCalledWith({});
+  });
+
+  it('should be able to add address', () => {
+    component.addAddress({});
+    expect(component.forceLoader).toBeTruthy();
+    expect(
+      checkoutDeliveryAddressFacade.createAndSetAddress
+    ).toHaveBeenCalledWith({});
+  });
+
+  it('should be able to get card content', () => {
+    const card = component.getCardContent(
+      mockAddress1,
+      undefined,
+      'default',
+      'shipTo',
+      'selected'
+    );
+    expect(card.title).toEqual('');
+    expect(card.textBold).toEqual('John Doe');
+    expect(card.text).toEqual([
+      'first line',
+      'second line',
+      'town, JP-27, JP',
+      'zip',
+      undefined,
+    ]);
+  });
+
+  describe('UI continue button', () => {
+    const getContinueBtn = () =>
+      fixture.debugElement.query(By.css('.cx-checkout-btns .btn-primary'));
+
+    it('should be disabled when no address is selected', () => {
+      component.selectedAddress = undefined;
+      fixture.detectChanges();
+      expect(getContinueBtn().nativeElement.disabled).toEqual(true);
+    });
+
+    it('should be enabled when address is selected', () => {
+      component.selectedAddress = mockAddress1;
+      fixture.detectChanges();
+      expect(getContinueBtn().nativeElement.disabled).toEqual(false);
+    });
+
+    it('should call "next" function after being clicked', () => {
+      spyOn(component, 'next');
+      component.selectedAddress = mockAddress1;
+      fixture.detectChanges();
+      getContinueBtn().nativeElement.click();
+      expect(component.next).toHaveBeenCalled();
+    });
+  });
+
+  describe('UI back button', () => {
+    const getBackBtn = () =>
+      fixture.debugElement
+        .queryAll(By.css('.btn-action'))
+        .find((el) => el.nativeElement.innerText === 'common.back');
+
+    it('should call "back" function after being clicked', () => {
+      spyOn(component, 'back').and.callThrough();
+      fixture.detectChanges();
+      getBackBtn()?.nativeElement.click();
+      expect(component.back).toHaveBeenCalled();
+    });
+  });
+
```

Comment:
Isn't this test for b2b?

Reply:
nay

Reply:
old test 

used to test the if condition for payment and otherwise the b2c normal one (which is normally credit card)

```
  describe('selectDefaultAddress', () => {
    describe('Account Payment', () => {
      it('should automatically select default shipping address when there is ONLY ONE', () => {
        isAccount.next(true);
        component.ngOnInit();
        component.selectDefaultAddress([mockAddress1], undefined);
        expect(component.selectAddress).toHaveBeenCalledWith(mockAddress1);
      });
    });

    describe('Credit Card Payment', () => {
      it('should automatically select default shipping address when there is no current selection', () => {
        component.doneAutoSelect = false;
        component.selectDefaultAddress(mockAddresses, undefined);
        expect(component.selectAddress).toHaveBeenCalledWith(mockAddress2);
      });
    });
  });
```

Reply:
Oh, you're right. Added, thank you :)